qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -158,29 +164,68 @@ def __init__(
 
```

Comment:
Could we cover the case where a user provides all argument positionally? Like
```suggestion
        if quantum_instance is not None and sampler is not None:
            raise ValueError("Only one of quantum_instance or sampler can be passed, not both!")

        # check positionally passing the sampler in the place of quantum_instance
        # which will be removed in future
        if isinstance(quantum_instance, BaseSampler):
            sampler = quantum_instance
            quantum_instance = None
```
It's an unlikely case to be used right now, but it already enables the future signature we will have 🙂